detox/test/ios/Podfile

```diff
@@ -29,4 +29,5 @@ end
 
 post_install do |installer|
   react_native_post_install(installer)
+  __apply_Xcode_12_5_M1_post_install_workaround(installer)
```

Comment:
We are not using Xcode 12.5 + M1 on our agents 😠

Reply:
This is part of the standard template, I worked so hard on that workaround 😅 - I'm not sure why 12.5+ (or 13+ now?) isn't in use in the agents? But if it is at all you will suffer immediate build failures without this workaround, and in fact there may be build failures just by including Swift, not just using 12.5+ or M1. Either way, it's part of the standard template and will also be removed in react-native 0.68 (0.67 still requires it unfortunately...). Building on Xcode right now is a real mess basically, and I would remove that line at your peril...

Reply:
Thanks for all your efforts on this upgrade Mike. I just saw the workaround details on RN. We are using Xcode 13, so it is relevant for our Xcode version. I was too quick to judge that it was relevant for v12.5 and not for the following versions (or specifically v13)..

I saw this error in [one of our builds](https://buildkite.com/wix-mobile-oss/detox/builds/126#98f5e3da-1543-43a6-a9d9-18e4ebe49178), and wondered why we need this call on our Podfile if we are not using M1 agents:
```
Generating Pods project
--
  | [!] An error occurred while processing the post-install hook of the Podfile.
  |  
  | undefined method `__apply_Xcode_12_5_M1_post_install_workaround' for #<Pod::Podfile:0x00007f7f5bd58f68>
  |  
  | /Users/builder/work/detox/test/ios/Podfile:32:in `block (2 levels) in from_ruby'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.10.2/lib/cocoapods-core/podfile.rb:179:in `post_install!'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:897:in `run_podfile_post_install_hook'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:885:in `block in run_podfile_post_install_hooks'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/user_interface.rb:145:in `message'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:884:in `run_podfile_post_install_hooks'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:329:in `block (2 levels) in create_and_save_projects'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer/xcode/pods_project_generator/pods_project_writer.rb:61:in `write!'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:328:in `block in create_and_save_projects'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/user_interface.rb:64:in `section'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:307:in `create_and_save_projects'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:299:in `generate_pods_project'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:178:in `integrate'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/installer.rb:166:in `install!'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/command/install.rb:52:in `run'
  | /Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/lib/cocoapods/command.rb:52:in `run'
  | /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.10.2/bin/pod:55:in `<top (required)>'
  | /usr/local/bin/pod:23:in `load'
  | /usr/local/bin/pod:23:in `<main>'
```

However, when [tried to remove this line](https://github.com/wix/Detox/pull/3146), our build failed on `xcodebuild` command..

Reply:
OK, the issue was that we are calling this method from RN64 build.

Reply:
Oh! Yes, that won't work :-), great catch